python_modules/dagster/dagster/core/execution/context/hook.py

```diff
@@ -367,6 +386,13 @@ def build_hook_context(
             with build_hook_context(resources={"foo": context_manager_resource}) as context:
                 hook_to_invoke(context)
     """
+    if op:
+        experimental_arg_warning("op", "build_hook_context")
+        return UnboundHookContext(
+            resources=check.opt_dict_param(resources, "resources", key_type=str),
+            mode_def=ModeDefinition,
+            solid=check.opt_inst_param(op, "op", (OpDefinition, PendingNodeInvocation)),
+        )
```

Comment:
I would maybe add an invariant s.t. someone can't provide both a solid and an op, just to distinguish that they're aliases a la `check.invariant(not (solid and op))`